### DISCUSSION TITLE
Training data set for interface systems

### DISCUSSION THREAD

**Discussion by namtranlab** [deepmodeling/deepmd-kit#1518] [Q&A] [answered]
Dear DeepMD experts,

I am trying to study the interface between a solvent and a surface. However, I am not sure how to generate the training set for DeepMD. 

1. My idea is to generate three main training sets though (AIMD). 1, a system for the bulk solvent. 2, a system for the bulk structure of the substrate. 3, a system of the interface between the two. Is it a reasonable way to train the DP for an interface system?

2. I would like to apply an external electric field to my system. I found that the DPLR and DW could be the answer to this task (https://arxiv.org/abs/2112.13327). However, I could not find any example or instruction on how to generate the training data or how the MD simulations were done in this case. I highly appreciate if anyone can provide some instruction on this topic.

Best,

**Comment by wanghan-iapcm** [comment, marked as answer]
1. The best way is to let the model tell what kind of training data is needed to improve the accuracy. This idea introduces the concurrent learning method Deep potential generator (DPGEN). You may refer to https://doi.org/10.1103/PhysRevMaterials.3.023804 and https://doi.org/10.1016/j.cpc.2020.107206 for more details. 
2. You may find the doc for dplr [here](https://github.com/deepmodeling/deepmd-kit/blob/3eb2550bf5a6b5a1897168e62b97f19afac7e64a/doc/model/dplr.md) and the literature https://arxiv.org/abs/2112.13327 .

**Reply by namtranlab** [reply]
Dear Dr. Wang

Thank you so much for your quick reply.
I will check your mentioned references.

Best

**Reply by namtranlab** [reply]
Dear Dr. Wang,

I would like to have one more question.

Can DeepMD handle data sets with different chemical compounds?
For example, my interface system has three types of elements [H, O, C].
My first data set is the surface with only one element [C].
The 2nd data set is the solvent [H, O].
And the 3rd data set has all of the three elements [H, O, C].

If it is possible, then how do I set the _type.raw_ and _type_map.raw_ in each data set and also the _"type_map"_ keyword in the training input.

Best

**Reply by wanghan-iapcm** [reply]
Sure, these data should be placed in different systems, other than sets in one system. Please check the data format used by deepmd-kit 
https://github.com/deepmodeling/deepmd-kit/blob/master/doc/data/index.md

**Reply by namtranlab** [reply]
Dear Dr. Wang

Thank you so much for your answer.
I got the idea now.

Best
